insights/parsers/sctp.py

```diff
@@ -254,3 +257,72 @@ def search(self, **args):
                 when args search do not match then it returns `[]`.
         """
         return keyword_search(self.data, **args)
+
+
+@parser(Specs.sctp_snmp)
+class SCTPSnmp(Parser, dict):
+    """
+    This parser parses the content of ``/proc/net/sctp/snmp`` file,
+    which contains statistics related to SCTP states, packets and chunks.
+
+    Sample content::
+
+        SctpCurrEstab                   	5380
+        SctpActiveEstabs                	12749
+        SctpPassiveEstabs               	55
+        SctpAborteds                    	2142
+        SctpShutdowns                   	5295
+        SctpOutOfBlues                  	36786
+        SctpChecksumErrors              	0
+        SctpOutCtrlChunks               	1051492
+
+    Data is stored in a dictionary.
+
+    Examples:
+
+        >>> type(sctp_snmp)
+        <class 'insights.parsers.sctp.SCTPSnmp'>
+        >>> sctp_snmp.get('SctpCurrEstab')
+        5380
+        >>> sctp_snmp.get('SctpChecksumErrors') == 0
+        True
+        >>> sctp_snmp.stats
+        {'SctpCurrEstab': 5380, 'SctpActiveEstabs': 12749, 'SctpPassiveEstabs': 55, 'SctpAborteds': 2142, 'SctpShutdowns': 5295, 'SctpOutOfBlues': 36786, 'SctpChecksumErrors': 0, 'SctpOutCtrlChunks': 1051492}
+
+    Resultant Data::
+
+        {
+            'SctpCurrEstab': 5380,
+            'SctpActiveEstabs': 12749,
+            'SctpPassiveEstabs': 55,
+            'SctpAborteds': 2142,
+            'SctpShutdowns': 5295,
+            'SctpOutOfBlues': 36786,
+            'SctpChecksumErrors': 0,
+            ...
+            ...
+        }
+
+    Raises:
+        SkipException: When contents are empty.
+        ParseException: When file contents are not in expected format.
+    """
+
+    def parse_content(self, content):
+        if (not content) or (not self.file_path):
+            raise SkipException("No Contents")
+
+        for line in content:
+            line_strip = line.split()
+            if len(line_strip) != 2:
+                raise ParseException("Contents are not compatible to this parser")
+
+            self[line_strip[0]] = int(line_strip[1]) if line_strip[1].isdigit() else None
+
+    @property
+    def stats(self):
+        """
+        dict: On success, it will return detailed sctp snmp stats in
+        a dictionary data format, on failure it will return ``None``
+        """
+        return {**self}
```

Comment:
Since the core is opensourced project and it could be run in all possible python versions (2.6, 2.7 and 3.6) environment. We need to support all of them. This `{**self}` is not supported by old python 2.6 and 2.7. Could you please change it to `self` directly?

Reply:
Removed the interface fucntion.